That was fun. So the upstream use of `libbsd` here isn't ideal (forgoing overlay mode in favour of explicit includes), and our `libbsd` package is also a bit broken on macOS right now. I think it would have been possible to just drop the dependency on `libbsd` for Darwin, but I went to the effort of fixing `libbsd` instead to avoid touching the source more than necessary. I'll open a PR for the `libbsd` changes, but here's the diff against the current PR head that it took to get this working; the changes should be harmless on Linux/glibc.

Additionally, I switched to using the CMake build system as it seems that upstream is using it for CI. Unfortunately it doesn't link `argp-standalone` like the Makefile does, so I had to add a patch for that. It should probably go upstream (and I hereby release it into the public domain and/or license it under [0BSD](https://opensource.org/license/0bsd/) to make that zero-friction in case I don't get around to it and you feel like doing so). I also changed the `argp-standalone` conditional to be a bit more precise for any Nix-on-musl-or-BSD weeaboos out there.

(I wonder if we should call this `mya` instead, to match the binary and AUR package name?)

```diff
diff --git a/pkgs/applications/misc/myanimelist-cli/argp.patch b/pkgs/applications/misc/myanimelist-cli/argp.patch
new file mode 100644
index 000000000000..352f8a077adc
--- /dev/null
+++ b/pkgs/applications/misc/myanimelist-cli/argp.patch
@@ -0,0 +1,26 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 273968c..236e5fb 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -3,10 +3,6 @@ project(mya)
+ set(CMAKE_C_STANDARD 11 )
+ set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -g -Wall")
+ 
+-if(UNIX)
+-	set(LINUX TRUE)
+-endif()
+-
+ set(SRC_DIR src)
+ set(INC_DIR include)
+ 
+@@ -17,7 +13,8 @@ set_target_properties(mya PROPERTIES RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DI
+ target_include_directories(mya PRIVATE ${INC_DIR})
+ 
+ set(LIBS curl json-c bsd)
+-if(LINUX)
+-    list(APPEND LIBS)
++find_library(ARGP argp)
++if(ARGP)
++    list(APPEND LIBS argp)
+ endif()
+ target_link_libraries(mya ${LIBS})
diff --git a/pkgs/applications/misc/myanimelist-cli/default.nix b/pkgs/applications/misc/myanimelist-cli/default.nix
index e4e15ff66b06..10aaaf39dda8 100644
--- a/pkgs/applications/misc/myanimelist-cli/default.nix
+++ b/pkgs/applications/misc/myanimelist-cli/default.nix
@@ -1,10 +1,12 @@
 { lib
 , stdenv
 , fetchFromGitHub
-, argp-standalone
+, cmake
+, ninja
 , curl
 , json_c
 , libbsd
+, argp-standalone
 }:
 
 stdenv.mkDerivation (finalAttrs: {
@@ -18,24 +20,31 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-EmdkPpYEUIk9hr6rbnixjvznKSEnTCSMZz/17BfHGCk=";
   };
 
+  nativeBuildInputs = [
+    cmake
+    ninja
+  ];
+
   buildInputs = [
     curl
     json_c
     libbsd
-  ] ++ lib.optionals stdenv.isDarwin [
+  ] ++ lib.optionals (!stdenv.hostPlatform.isGnu) [
     argp-standalone
   ];
 
-  # support LLVM and such
-  postPatch = ''
-    substituteInPlace Makefile \
-      --replace 'CC:=gcc' 'CC:=${stdenv.cc.targetPrefix}cc'
-  '';
+  patches = [
+    ./argp.patch
+  ];
 
   installPhase = ''
     runHook preInstall
-    mkdir -p $out
+
+    # Based on the upstream PKGBUILD
+    mkdir -p $out/share/doc/${finalAttrs.pname}
     cp -a bin $out
+    cp $cmakeDir/README.md $out/share/doc/${finalAttrs.pname}
+
     runHook postInstall
   '';
 
@@ -54,7 +63,5 @@ stdenv.mkDerivation (finalAttrs: {
     maintainers = with maintainers; [ pbsds ];
     mainProgram = "mya";
     platforms = platforms.all;
-    # FIXME: darwin maintainer needed
-    #broken = stdenv.isDarwin; # src/mya.c:4:10: fatal error: 'argp.h' file not found
   };
 })
```

--- a/pkgs/applications/misc/mya/argp.patch
+++ b/pkgs/applications/misc/mya/argp.patch
@@ -1,0 +1,26 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 273968c..236e5fb 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -3,10 +3,6 @@ project(mya)
+ set(CMAKE_C_STANDARD 11 )
+ set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -g -Wall")
+ 
+-if(UNIX)
+-	set(LINUX TRUE)
+-endif()
+-
+ set(SRC_DIR src)
+ set(INC_DIR include)
+ 
+@@ -17,7 +13,8 @@ set_target_properties(mya PROPERTIES RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DI
+ target_include_directories(mya PRIVATE ${INC_DIR})
+ 
+ set(LIBS curl json-c bsd)
+-if(LINUX)
+-    list(APPEND LIBS)
++find_library(ARGP argp)
++if(ARGP)
++    list(APPEND LIBS argp)
+ endif()
+ target_link_libraries(mya ${LIBS})

--- a/pkgs/applications/misc/mya/default.nix
+++ b/pkgs/applications/misc/mya/default.nix
@@ -1,0 +1,67 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, ninja
+, curl
+, json_c
+, libbsd
+, argp-standalone
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "mya";
+  version = "1.0.0";
+
+  src = fetchFromGitHub {
+    owner = "jmakhack";
+    repo = "myanimelist-cli";
+    rev = "refs/tags/v${finalAttrs.version}";
+    hash = "sha256-EmdkPpYEUIk9hr6rbnixjvznKSEnTCSMZz/17BfHGCk=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    ninja
+  ];
+
+  buildInputs = [
+    curl
+    json_c
+    libbsd
+  ] ++ lib.optionals (!stdenv.hostPlatform.isGnu) [
+    argp-standalone
+  ];
+
+  patches = [
+    ./argp.patch
+  ];
+
+  installPhase = ''
+    runHook preInstall
+
+    # Based on the upstream PKGBUILD
+    mkdir -p $out/share/doc/${finalAttrs.pname}
+    cp -a bin $out
+    cp $cmakeDir/README.md $out/share/doc/${finalAttrs.pname}
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "Minimalistic command line interface for fetching user anime data from MyAnimeList";
+    longDescription = ''
+      Minimalistic command line interface for fetching user anime data from MyAnimeList.
+
+      You have to run this with the MYANIMELIST_CLIENT_ID environ variable set.
+      Where to get one: <https://myanimelist.net/apiconfig>.
+      Select the type `other`.
+    '';
+    homepage = "https://github.com/jmakhack/myanimelist-cli";
+    changelog = "https://github.com/jmakhack/myanimelist-cli/releases/tag/v${finalAttrs.version}";
+    license = licenses.mit;
+    maintainers = with maintainers; [ pbsds ];
+    mainProgram = "mya";
+    platforms = platforms.all;
+  };
+})

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -951,6 +951,8 @@ with pkgs;
 
   mongosh = callPackage ../development/tools/mongosh { };
 
+  mya = callPackage ../applications/misc/mya { };
+
   mysql-shell = callPackage ../development/tools/mysql-shell {
     inherit (darwin) cctools developer_cmds DarwinTools;
     inherit (darwin.apple_sdk.frameworks) CoreServices;

